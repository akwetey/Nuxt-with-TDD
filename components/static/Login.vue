<template>
  <div class="container">
    <div class="card px-5 py-4 shadow-sm">
      <div class="text-center mb-2">
        <h4 class="text-uppercase mt-0">Sign In</h4>
      </div>
      <form @submit.prevent="login">
        <div class="form-group mb-3">
          <label for="emailaddress" class="float-left">Email</label>
          <input
            class="form-control"
            type="email"
            id="emailaddress"
            v-model="email"
            required
            placeholder="Enter your email"
          />
        </div>

        <div class="form-group mb-3">
          <label for="password" class="float-left">Password</label>
          <input
            class="form-control"
            type="password"
            v-model="password"
            required
            id="password"
            placeholder="Enter your password"
          />
        </div>
        <div class="form-group mb-0 text-center mb-3">
          <button
            class="btn btn-primary btn-block"
            type="submit"
            id="loginBtn"
            style="background-color:#409df7; border-color:#409df7"
            ref="loginBtn"
          >
            Log In
          </button>
        </div>
        <div class="d-flex">
          <div>
            <nuxt-link to="/organisation/forgot-password"
              >Forgot Password</nuxt-link
            >
          </div>
          <div class="ml-auto">
            <nuxt-link to="/organisation/register"
              >Don't have an account? Register</nuxt-link
            >
          </div>
        </div>
      </form>
    </div>
  </div>
</template>

<script>
import { mapGetters, mapActions } from "vuex";
export default {
  name: "Login",
  data() {
    return {
      email: "",
      password: "",
      showDismissibleAlert: false
    };
  },
  methods: {
    ...mapActions(["auth"]),
    login() {
      const btn = this.$refs.loginBtn;
      console.log(this.email, this.password, btn);
    }
  },
  computed: mapGetters(["message"])
};
</script>

<style scoped>
.card {
  border-color: #edf2f9;
}
.container {
  margin: 100px auto;
  overflow: hidden;
  width: 400px;
  position: relative;
  min-height: 50vh;
}
</style>
