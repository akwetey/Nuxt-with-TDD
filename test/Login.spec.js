import { shallowMount, createLocalVue } from "@vue/test-utils";
import Login from "@/components/static/Login.vue";
import Vuex from "vuex";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Login form", () => {
  test("calls login action", () => {
    const loginMock = jest.fn(() => Promise.resolve());
    const store = new Vuex.Store({
      actions: {
        auth: loginMock
      }
    });
    const wrapper = shallowMount(Login, { localVue, store });
    wrapper.find("button").trigger("click");
    expect(loginMock).toHaveBeenCalled();
  });

  test("Form should render correctly", () => {
    const wrapper = shallowMount(Login, {
      stubs: {
        NuxtLink: true
      }
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
