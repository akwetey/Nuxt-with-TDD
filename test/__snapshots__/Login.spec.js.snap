// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login form Form should render correctly 1`] = `
"<div class=\\"container\\">
  <div class=\\"card px-5 py-4 shadow-sm\\">
    <div class=\\"text-center mb-2\\">
      <h4 class=\\"text-uppercase mt-0\\">Sign In</h4>
    </div>
    <form>
      <div class=\\"form-group mb-3\\"><label for=\\"emailaddress\\" class=\\"float-left\\">Email</label> <input type=\\"email\\" id=\\"emailaddress\\" required=\\"required\\" placeholder=\\"Enter your email\\" class=\\"form-control\\"></div>
      <div class=\\"form-group mb-3\\"><label for=\\"password\\" class=\\"float-left\\">Password</label> <input type=\\"password\\" required=\\"required\\" id=\\"password\\" placeholder=\\"Enter your password\\" class=\\"form-control\\"></div>
      <div class=\\"form-group mb-0 text-center mb-3\\"><button type=\\"submit\\" id=\\"loginBtn\\" class=\\"btn btn-primary btn-block\\" style=\\"background-color: rgb(64, 157, 247); border-color: #409df7;\\">
          Log In
        </button></div>
      <div class=\\"d-flex\\">
        <div>
          <nuxtlink-stub to=\\"/organisation/forgot-password\\">Forgot Password</nuxtlink-stub>
        </div>
        <div class=\\"ml-auto\\">
          <nuxtlink-stub to=\\"/organisation/register\\">Don't have an account? Register</nuxtlink-stub>
        </div>
      </div>
    </form>
  </div>
</div>"
`;
